torch/csrc/lazy/python/init.cpp

```diff
@@ -17,6 +20,12 @@
 namespace torch {
 namespace lazy {
 
+struct NoGilSection {
```

Comment:
we don't want to land NoGilSection, can you switch to `pybind11::gil_scoped_release no_gil;` like mark_step uses?